### ISSUE TITLE
Obscure exception

### ISSUE THREAD

**Issue by brianfromoregon** [jenetics/jenetics#45]
Hi,

I've likely misconfigured something, but even so there should be a more friendly/helpful exception explaining the problem.

```
Exception in thread "main" java.util.concurrent.CompletionException: java.lang.ArrayIndexOutOfBoundsException: 1
	at java.util.concurrent.CompletableFuture.internalComplete(CompletableFuture.java:205)
	at java.util.concurrent.CompletableFuture$AsyncApply.exec(CompletableFuture.java:507)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:902)
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1689)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1644)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
	at org.jenetics.Crossover.recombine(Crossover.java:69)
	at org.jenetics.Recombinator.lambda$alter$42(Recombinator.java:110)
	at org.jenetics.Recombinator$$Lambda$58/2026408975.applyAsInt(Unknown Source)
	at java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:210)
	at java.util.stream.IntPipeline$4$1.accept(IntPipeline.java:250)
	at java.util.stream.IntPipeline$9$1.accept(IntPipeline.java:344)
	at java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:110)
	at java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:693)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.IntPipeline.reduce(IntPipeline.java:456)
	at java.util.stream.IntPipeline.sum(IntPipeline.java:414)
	at org.jenetics.Recombinator.alter(Recombinator.java:111)
	at org.jenetics.CompositeAlterer.lambda$alter$8(CompositeAlterer.java:78)
	at org.jenetics.CompositeAlterer$$Lambda$55/792586270.applyAsInt(Unknown Source)
	at java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:210)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.IntPipeline.reduce(IntPipeline.java:456)
	at java.util.stream.IntPipeline.sum(IntPipeline.java:414)
	at org.jenetics.CompositeAlterer.alter(CompositeAlterer.java:79)
	at org.jenetics.engine.Engine.alter(Engine.java:399)
	at org.jenetics.engine.Engine.lambda$evolve$121(Engine.java:269)
	at org.jenetics.engine.Engine$$Lambda$35/1413653265.apply(Unknown Source)
	at org.jenetics.engine.TimedResult.lambda$of$114(TimedResult.java:81)
	at org.jenetics.engine.TimedResult$$Lambda$38/1418621776.apply(Unknown Source)
	at java.util.concurrent.CompletableFuture$AsyncApply.exec(CompletableFuture.java:501)
	... 5 more
```

the line that throws is

		final Phenotype<G, C> pt2 = population.get(individuals[1]);

This is with version 3.1

**Comment by brianfromoregon** [comment]
This code shows a reproduction: https://gist.github.com/anonymous/6bdd958513bff3352b19
If I increase population size to >=3 then it works.

**Comment by jenetics** [comment]
In #34 and #35 I added special handling for small population. This should prevent the given exception.

**Comment by jenetics** [comment]
Has been fixed in #34 and #35.
